### ISSUE TITLE
formatter moves `type:ignore` comment and makes line too long

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#7730]
# before
```py
def run_pytest(pytester, subprocess):
    result = (
        pytester.runpytest_subprocess 
        if subprocess else pytester.runpytest  # type:ignore[no-any-expr]
    )(*(pytest_args or []))
```
# after
```py
def run_pytest(pytester, subprocess):
    result = (
        pytester.runpytest_subprocess if subprocess else pytester.runpytest  # type:ignore[no-any-expr]
    )(*(pytest_args or []))

```
```
E501 Line too long (103 > 100 characters)
```
# playground
https://play.ruff.rs/720b0769-e4d7-430f-8e14-5c7295607d6d

**Comment by charliermarsh** [comment]
This is an intentional behavior as we exclude pragmas when computing line length in the formatter: https://github.com/astral-sh/ruff/blob/main/crates/ruff_python_formatter/README.md#pragma-comments-are-ignored-when-computing-line-width.

In the next release, we're planning to ignore them when enforcing E501 violations to make the linter consistent with this behavior.

**Comment by charliermarsh** [comment]
I'm gonna close as fixed by https://github.com/astral-sh/ruff/pull/7692, but won't actually be fixed until we cut the next release.
